demos/https/iot_demo_https_async.c

```diff
@@ -171,7 +172,7 @@ typedef struct _asyncDownloadData {
  * Since all of the requests in this demo are on the same connection, only one async task at a time will be accessing 
  * this buffer to retrieve data from the network to print. 
  */
-static uint8_t _pRespBodyBuffer[IOT_DEMO_HTTPS_RESP_BODY_BUFFER_SIZE] = { 0 };
+static char _pRespBodyBuffer[IOT_DEMO_HTTPS_RESP_BODY_BUFFER_SIZE] = { 0 };
```

Comment:
I will make the readResponseBody buffer take a uint8_t* pointer for possible binary body. let me know that works.